### PR TITLE
Enable cache action for lint workflow

### PR DESCRIPTION
**Pull request by huydhn**
Cache all python dependencies using [GHA cache](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows).  I'm doing this for lint workflow first and will slowly roll it out to other workflows.

### Testing

Before caching, pip cache is not found. Dependencies installation continues as usual:

Screen Shot 2022-08-24 at 16 36 15

After caching https://github.com/pytorch/pytorch/runs/8006214772?check_suite_focus=true. The long hash at the end of the cache key is the hash of requirements files

Screen Shot 2022-08-24 at 16 51 51

Note that the cache is in the runners themselves. This should be a transparent process.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -14,17 +14,19 @@ jobs:
...
         with:
           python-version: 3.8
           architecture: x64
 
-      - name: Checkout PyTorch
-        uses: pytorch/pytorch/.github/actions/checkout-pytorch@master
-        with:
-          submodules: false
-
+      # Lintrunner is not cached because it might change frequently
```
Comment (suo) [inline .github/workflows/lint.yml:29]: it's probably okay to cache lintrunner. We haven't need to release a new version in a while https://github.com/suo/lintrunner/commits/main

**Review by suo — COMMENTED** [review]
what's the expected impact on this change on TTS?

**Comment by huydhn** [comment]
> what's the expected impact on this change on TTS?

Not much I think because installing dependencies isn't the long pole in term of TTS.  However, it has a minor boon that allows the cached copied to be used whenever there are upstream issues with pip.  And the setup is pretty easy as you can see in the PR

**Review by seemethere — APPROVED** [review]

**Review by suo — APPROVED** [review]
coolio

**Review by malfet — APPROVED** [review]

**Comment by malfet** [comment]
Looks good, but please avoid images in PR description, rather share link to respective runs + text-readable perf numbers

**Comment by malfet** [comment]
One needs to be careful with introducing new dependencies, as lint will fail for all users that have not rebased, as workflow definition is checked from master, but rest of the code is from PR, which results in the following failures:
https://github.com/pytorch/pytorch/runs/8020179452?check_suite_focus=true

**Comment by huydhn** [comment]
> One needs to be careful with introducing new dependencies, as lint will fail for all users that have not rebased, as workflow definition is checked from master, but rest of the code is from PR, which results in the following failures: https://github.com/pytorch/pytorch/runs/8020179452?check_suite_focus=true

Note. In hindsight, I should have added the new file first, then made the change.

**Pull request closed without merging**